modules/extensions/partials/proc_creating-a-telemetry-plugin-for-devworkspaces.adoc

```diff
@@ -0,0 +1,423 @@
+* Specifying of a location of a custom plug-in to {prod-short} by setting the `workspacesDefaultPlugins` attribute from the `CheCluster` custom resource.
+
+== Getting started
+
+This document describes the steps required to extend the {prod-short} telemetry system to communicate with to a custom back-end:
+
+. Creating a server process that receives events
+. Extending {prod-short} libraries to create a back-end that send events to the server
+. Packaging the telemetry back-end in a container and deploying it to an image registry
+. Adding a plug-in for your back-end and instructing {prod-short} to load the plug-in in your {devworkspace}s
+
+A finished example of the telemetry back-end is available link:https://github.com/che-incubator/devworkspace-telemetry-example-plugin[here].
+
+[discrete]
+== Creating a server that receives events
+
+For demonstration purposes, this example shows how to create a server that receives events from our telemetry plug-in and writes them to standard output.
+
+For production use cases, consider integrating with a third-party telemetry system (for example, Segment, Woopra) rather than creating your own telemetry server. In this case, use your provider's APIs to send events from your custom back-end to their system.
+
+The following Go code starts a server on port `8080` and writes events to standard output:
+
+.`main.go`
+====
+[source,go]
+----
+include::example$creating-a-telemetry-plug-in-for-devworkspaces/main.go[]
+----
+====
+
+Create a container image based on this code and expose it as a deployment in OpenShift in the {prod-namespace} {orch-namespace}. The code for the example telemetry server is available at link:https://github.com/che-incubator/telemetry-server-example[telemetry-server-example]. To deploy the telemetry server, clone the repository and build the container:
+
+----
+$ git clone https://github.com/che-incubator/telemetry-server-example
+$ cd telemetry-server-example
+$ docker build -t registry/organization/telemetry-server-example:latest .
+$ docker push registry/organization/telemetry-server-example:latest
+----
+
+Both `manifest_with_ingress.yaml` and `manifest_with_route` contain definitions for a Deployment and Service. The former also defines a {kubernetes} Ingress, while the latter defines also an OpenShift Route.
+
+In the manifest file, replace the `image` and `host` fields to match the image you pushed, and the public hostname of your {platforms-name} cluster. Then run:
+
+[subs="+quotes"]
+----
+$ kubectl apply -f manifest_with_[ingress|route].yaml -n {prod-namespace}
+----
+
+== Creating the back-end project
+
+NOTE: For fast feedback when developing, it is recommended to do development inside a {devworkspace}. This way, you can run the application in a cluster and receive events from the front-end telemetry plug-in.
+
+pass:[<!-- vale RedHat.TermsErrors = NO -->]
+
+. Maven Quarkus project scaffolding:
++
+----
+$ mvn io.quarkus:quarkus-maven-plugin:2.7.1.Final:create \
```

Comment:
@themr0c, is there a way to pass through the contents of a code block so that Vale does not throw an error?
![image](https://user-images.githubusercontent.com/78564304/158609627-d356d42f-37b8-458c-a554-3242e0d8e0e3.png)